practitioner-service/src/test/java/uk/ac/ox/ndph/mts/practitioner_service/repository/FhirContextWrapperTest.java

```diff
@@ -3,19 +3,159 @@
 import ca.uhn.fhir.context.FhirContext;
 import ca.uhn.fhir.rest.server.exceptions.BaseServerResponseException;
 import org.assertj.core.api.Assertions;
+import org.hl7.fhir.instance.model.api.IBaseResource;
 import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Practitioner;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.RestException;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+
 class FhirContextWrapperTest {
 
+    final FhirContextWrapper sut = new FhirContextWrapper();
+    final Bundle EMPTY_BUNDLE = new Bundle();
+    final Bundle bundle = new Bundle();
+    final Logger SILENT_LOGGER = mock(Logger.class);
+
+//    @Test
+//    void toListOfResources_whenBundleHasNoResources_logNpeThrownByHapi() {
+//        // Arrange
+//        FhirContext fhirContext = mock(FhirContext.class);
+//        Logger logger = mock(Logger.class);
+//        FhirContextWrapper sut = new FhirContextWrapper(fhirContext, logger);
+//
+//        // Act
+//        sut.toListOfResources(EMPTY_BUNDLE);
+//
+//        // Assert
+//        verify(logger).error(eq("Failed to fetch resources from bundle"), any(NullPointerException.class));
+//    }
+
+//    @Test
+//    void toListOfResources_whenBundleHasNoResources_returnEmptyList() {
+//        // Arrange
+//        FhirContext fhirContext = mock(FhirContext.class);
+//        FhirContextWrapper sut = new FhirContextWrapper(fhirContext);
+//
+//        // Act
+//        final List<IBaseResource> resources = sut.toListOfResources(EMPTY_BUNDLE);
+//
+//        // Assertions
+//        assertThat(resources).isEmpty();
+//    }
+
+    @Test
+    void getResourcesFrom_whenNullBundle_thenThrowException() {
```

Comment:
I also noticed that you changed the naming convention for the tests. If we're going to do that it should be an informed decision and then some work to align other classes & services across the system.